music21/stream/iterator.py

```diff
@@ -533,7 +533,7 @@ def matchingElements(self):
         savedRestoreActiveSites = self.restoreActiveSites
         self.restoreActiveSites = True
 
-        me = [x for x in self]
+        me = [x for x in self]  # pylint: disable=unnecessary-comprehension
```

Comment:
or just list(self)

Reply:
Tried it and caused some nasty infinite loops:
```
..EEE..F....FF..FF.FFFF.FFFFFFFF.E..F.FFFFFFFFF.FFFFFFF.FFFFFFFFF..FFFF.FFFFFFEEEEEEEEEE.E..EEEEEEEEEEEEEEEEEEEE.EEEEEEEEE.E..EEEEEE.EEEE.EEE..EEEEEEEE...EEEEEEEEEEEE.EEEEEEEEEEEEEEEEEEEEEEEEEEEEE..EEEEEEEEEE.EEEEE...EEEEEEE...EEE.EEE.E.EEEEE....EEEE.E.EE..EE.EEEEEE..EEEEE.E...E.EEEEEEEEEEEEEEE.EEEEEE.EEEEEEE.FFFF.....EE.FFFFFFFFF....FFF..F..FFFFF.FFFFFF..FFFFF.F...FF..F.F.FFFFF.F..F.FFF...FFFF.F...FFFF.FFFF....FFFFFF.F..F......FF...FFFFFFFFFF...FFFF.FFF.F..FF...FF.F..F....FF.FFFFF.FFFF.FFFFFF..F....FFF.FFFFFF...E.EF..F.....EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE.E.......FFFFFF.F.FFFFFF.FFF.............E.EE.EE.EEEEEEEEE.EEEEEEEEEEEEEEEEEE.EEEEEEEEEEEE.EEEE..........F...........................FFFFF..FF.........................................................................................EEE........E....................................................................E............................................................................................................EE.....EEE...E......EE.E....EEEEE.EE.............FFFFFF...F.......F.F............EEEEEEEEEEEF.FF...................EEE.....FFF.F.....F..FF........FE.F......................................F...................EEE...F......F...............................F.F................................F..............FFFFFFFFF.FF.FFFFFF.FFF.F.F...F.FFFF..F..FEEFFF..FFF...FFFF..FFFFF.FEFFF..............E......F....EEEEEEEE......EEE....E..................................................................EF........FF.F.....E.EE.EEE.EEEE.EEE.EEEE........F........F....F.........................E........................................................E........................Failed on parse/index for, bach/bwv190.7-inst.mxl: maximum recursion depth exceeded while calling a Python object
Failed on parse/index for, bach/bwv190.7.mxl: maximum recursion depth exceeded while calling a Python object
FFFailed on parse/index for, bach/bwv197.5.mxl: maximum recursion depth exceeded while calling a Python object
Failed on parse/index for, bach/bwv190.7.mxl: maximum recursion depth exceeded while calling a Python object
Failed on parse/index for, bach/bwv197.10.mxl: maximum recursion depth exceeded while calling a Python object
FFFFF.F.FFFFFFFFF.F...FFF..............................................................E....................EEE.....................................................................................................EEEEEEEEEEEEEEEEF.....F......................................................F....F.....F...............E...............................EE........E...................................................................................
```

Sorry for running these tests on Travis, btw, I'm on a new iMac and experimenting with NOT using a venv and there are some things I haven't worked out.

Reply:
Not at all -- run on Travis as much as you'd like!   I'll try to make sure to put a note here to say "cannot be replaced by list(x)" -- presumably because list() uses this function.